### ISSUE TITLE
Detect collisions for sleeping body

### ISSUE THREAD

**Issue by jakubgruber** [liabru/matter-js#409]
Hello @liabru, I am using Matter JS for school project that rewrites trickytowers game in JS. 
I have a static body at the bottom and I generate block, that moves down every N seconds. This block needs to be placed on top of the platform. 

Currently, I set block as a sleeping and I want to wake it up once it collides with other objects or sits down on the platform. 

It is working only if I drop a block on top of this. Then it changes status from isSleeping to non sleeping and new block is created. But if my 'active' shape that needs to be placed sits on top of any other block, it does not wake up. It is also not detected by sensors (which was written in doc, I know)

Do you have any recommendation how to do that, please?

To sum up what I need to do is that I need to add shape into Matter World, so it is rendered, but I want to manage its move and position by mysleft unless it hits any other object.

Demo can be visible on https://jakubgruber.gitlab.io/

Thank you

**Comment by liabru** [comment]
I'm not quite sure I follow but have you tried using [collision events](http://brm.io/matter-js/demo/#events)?

**Comment by jakubgruber** [comment]
I am using collision events, but that actually does not solve the problem. I need to be able to move with the body (it means gravity is not applied), but the body is already added to Matter.World and I need to detect collisions. It does not detect collision on sleeping body.

**Comment by liabru** [comment]
If I understand correctly, you're setting them to sleep and then moving them? Instead what you should do is set them as static, move them and then set them dynamic again. Take a look at the [Body Manipulation demo](http://brm.io/matter-js/demo/#manipulation) which shows how to do this.
